### ISSUE TITLE
SOCKS support

### ISSUE THREAD

**Issue by ilf** [irssi/irssi#915]
As discussed with @ahf, native [SOCKS](https://en.wikipedia.org/wiki/SOCKS) support would be awesome to use Tor without [pain](https://trac.torproject.org/projects/tor/wiki/doc/TorifyHOWTO/irssi).

**Comment by ailin-nemui** [comment]
I recommend https://github.com/rofl0r/proxychains-ng

here is a starter: #148

**Comment by ilf** [comment]
@ailin-nemui Thanks for #148, I missed that one, because I searched only for "SOCKS". Sorry.

Besides proxychains-ng there is also [torsocks](https://gitweb.torproject.org/torsocks.git), but both are "hacks", not native.

**Comment by bmeh** [comment]
Yeah, I'm using torsocks + firejail with irssi: `firejail --profile=~/.config/firejail/irssi.profile torsocks -i /usr/bin/irssi`, but I would not consider adding an alias painful. :P

Still, native-anything support would be awesome, in general, unless it makes irssi heavily bloated.
Also, what does `--with-socks` do? It says "Build with socks support". Is that what you need, does it work, or not?

**Comment by bmeh** [comment]
Yeah, I'm using torsocks + firejail with irssi: `firejail --profile=~/.config/firejail/irssi.profile torsocks -i /usr/bin/irssi`, but I would not consider adding an alias painful. :P

Still, native-anything support would be awesome, in general, unless it makes irssi heavily bloated.
Also, have you tried to build with `--with-socks` It says "Build with socks support". Is that what you need, does it work, or not?

**Comment by ilf** [comment]
@bmeh: Thanks for the hint about `--with-socks`, I also didn't know about that. It seems to be [from 1999](https://github.com/irssi/irssi/blob/master/NEWS#L2867) and [depend](https://github.com/irssi/irssi/blob/master/docs/startup-HOWTO.txt#L578) on [dante](https://www.inet.no/dante/).

I have no idea what state that in. From https://github.com/irssi/irssi/pull/148/commits/bb276c0b8023ae31805b06ba5e993125b2a27e28 and https://github.com/irssi/irssi/pull/148 I get the impression that `--with-socks` is very stale - and my whish is already being worked on, so it can be closed as dupe.

**Comment by dequis** [comment]
Keep this open, the dante thing is terrible and there's no open issue tracking this. We have a bunch of workarounds (dante counts as one imo), but no decent builtin solution for this problem. And #148 is orphan, meaning not actually being worked on.

**Comment by ailin-nemui** [comment]
last time I checked, dante (--with-socks) didn't support SSL

**Comment by ailin-nemui** [comment]
you can test #1603
